google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/it/env/DirectPathEnv.java

```diff
@@ -0,0 +1,182 @@
+        isDirectPathEnabled(),
+        getRequiredProperty(PROJECT_PROPERTY_NAME),
+        getRequiredProperty(INSTANCE_PROPERTY_NAME),
+        getRequiredProperty(TABLE_PROPERTY_NAME));
+  }
+
+  public DirectPathEnv(
+      boolean directPathEnabled, String projectId, String instanceId, String tableId) {
+    this.directPathEnabled = directPathEnabled;
+    this.projectId = projectId;
+    this.instanceId = instanceId;
+    this.tableId = tableId;
+    this.rowPrefix = UUID.randomUUID() + "-";
+  }
+
+  @Override
+  public void start() throws IOException {
+    BigtableDataSettings.Builder settingsBuilder =
+        BigtableDataSettings.newBuilder().setProjectId(projectId).setInstanceId(instanceId);
+
+    // Direct path test environment uses a different endpoint.
+    settingsBuilder.stubSettings().setEndpoint(DIRECT_PATH_SERVICE_NAME + ":443");
+
+    // TODO(igorbernstein): move direct path conditional logic to gax
+    // Direct path environment can be accessed via CFE or direct path. When using direct path,
+    // disable connection pooling.
+    if (directPathEnabled) {
+      settingsBuilder
+          .stubSettings()
+          .setTransportChannelProvider(
+              EnhancedBigtableStubSettings.defaultGrpcTransportProviderBuilder()
+                  .setPoolSize(1)
+                  .build());
+    }
+
+    dataClient = BigtableDataClient.create(settingsBuilder.build());
+  }
+
+  @Override
+  public void stop() throws Exception {
+    deleteRows();
+    dataClient.close();
+  }
+
+  @Override
+  public BigtableDataClient getDataClient() {
+    return dataClient;
+  }
+
+  @Override
+  public String getProjectId() {
+    return projectId;
+  }
+
+  @Override
+  public String getInstanceId() {
+    return instanceId;
+  }
+
+  @Override
+  public String getTableId() {
+    return tableId;
+  }
+
+  @Override
+  public String getFamilyId() {
+    return FAMILY_ID;
+  }
+
+  @Override
+  public String getRowPrefix() {
+    return rowPrefix;
+  }
+
+  private void deleteRows() throws InterruptedException, ExecutionException, TimeoutException {
+    Query query = Query.create(tableId).prefix(rowPrefix);
+
+    List<ApiFuture<Void>> futures = Lists.newArrayList();
+    ServerStream<Row> rows = dataClient.readRows(query);
+    for (Row row : rows) {
+      ApiFuture<Void> future =
+          dataClient.mutateRowAsync(RowMutation.create(tableId, row.getKey()).deleteRow());
+      futures.add(future);
+    }
+
+    ApiFutures.allAsList(futures).get(10, TimeUnit.MINUTES);
+  }
+
+  private static String getRequiredProperty(String prop) {
+    String value = System.getProperty(prop);
+    if (value == null || value.isEmpty()) {
+      throw new RuntimeException("Missing system property: " + prop);
+    }
+    return value;
+  }
+
+  // TODO(igorbernstein): move direct path conditional logic to gax
+  private static boolean isDirectPathEnabled() {
+    String whiteList = System.getenv(DIRECT_PATH_ENV_VAR);
```

Comment:
minor: would rename this to directPathEnvVar for consistency

Reply:
I'd prefer to keep as is to keep it consistent with gax:
https://github.com/googleapis/gax-java/blob/61f236d3b7dcdbaf65217d37fd23c8686126f51b/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java#L200-L207